src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/solvers/IceFillSolver.kt

```diff
@@ -51,50 +54,65 @@ object IceFillSolver {
-                                            direction.rotateYCCW()
-                                        } else if (checkChestTorches(direction.rotateYCCW())) {
-                                            direction.rotateY()
-                                        } else continue
+                                        val offsetDir = listOf(direction.rotateYCCW(), direction.rotateY()).find {
+                                            return@find world.getBlockState(
+                                                pos.offset(
+                                                    it,
+                                                    1
+                                                )
+                                            ).block == Blocks.torch && world.getBlockState(
+                                                pos.offset(
+                                                    it.opposite,
+                                                    3
+                                                )
+                                            ).block == Blocks.torch
+                                        }?.opposite ?: continue
 
                                         if (world.getBlockState(
                                                 pos.offset(direction.opposite)
                                                     .offset(offsetDir)
                                                     .down(2)
                                             ).block == Blocks.stone_brick_stairs) {
+                                            //chestCenter: -11 75 -89; direction: east
+                                            val chestCenter = pos.offset(offsetDir)
+
+                                            val starts = Triple(
+                                                //three: -33 70 -89
+                                                chestCenter.down(5).offset(direction.opposite, 22),
+                                                //five: -28 71 -89
+                                                chestCenter.down(4).offset(direction.opposite, 17),
+                                                //seven: -21 72 -89
+                                                chestCenter.down(3).offset(direction.opposite, 10),
+                                            )
+                                            val ends = Triple(
+                                                //three: -29 70 -89
+                                                starts.first.offset(direction, 3),
+                                                //five: -23 71 -89
+                                                starts.second.offset(direction, 5),
+                                                //seven: -14 72 -89
+                                                starts.third.offset(direction, 7),
+                                            )
```

Comment:
Since it's been months when I originally added it, idk exactly how much the performance increased, however it was definitely noticable from what I remember. As before it scanned a 51x51x3 area, so about 7803 blocks and that times 2 (So 15606 blocks), since it also scans all the blocks below it. You could probably optimize the original code enough for it to be usable but I couldn't be bothered.

Reply:
Alright, just checking since hard coding can mean that updates will break it in the future. Personally I'd prefer scanning (definitely with a more optimized area) over hard coding but if it works, that's fine.